service/src/test/java/bio/terra/cbas/controllers/TestRunSetsApiController.java

```diff
@@ -148,4 +154,109 @@ void chainCallsTogether() throws Exception {
         newRunCaptor.getValue().submissionTimestamp(),
         greaterThan(OffsetDateTime.now().minus(Duration.ofSeconds(60))));
   }
+
+  @Test
+  void maximumRecordIds() throws Exception {
+
+    final String workflowUrl = "www.example.com/wdls/helloworld.wdl";
+    final String recordType = "MY_RECORD_TYPE";
+    final String recordId1 = "MY_RECORD_ID_1";
+    final String recordId2 = "MY_RECORD_ID_2";
+    final String recordAttribute = "MY_RECORD_ATTRIBUTE";
+    final int recordAttributeValue = 100;
+    RecordAttributes recordAttributes = new RecordAttributes();
+    recordAttributes.put(recordAttribute, recordAttributeValue);
+    final String cromwellWorkflowId = UUID.randomUUID().toString();
+    final String outputDefinitionAsString =
+        """
+        [ {
+          "output_name" : "myWorkflow.myCall.outputName1",
+          "output_type" : "String",
+          "record_attribute" : "foo_rating"
+        } ]""";
+
+    // Set up API responses:
+    when(wdsService.getRecord(recordType, recordId1))
+        .thenReturn(
+            new RecordResponse().type(recordType).id(recordId1).attributes(recordAttributes));
+    when(wdsService.getRecord(recordType, recordId2))
+        .thenReturn(
+            new RecordResponse().type(recordType).id(recordId2).attributes(recordAttributes));
+
+    when(cromwellService.submitWorkflow(eq(workflowUrl), any()))
+        .thenReturn(new RunId().runId(cromwellWorkflowId));
+
+    String request =
+        requestTemplate.formatted(
+            workflowUrl,
+            outputDefinitionAsString,
+            recordType,
+            "[ \"%s\", \"%s\" ]".formatted(recordId1, recordId2));
+
+    MvcResult result =
+        mockMvc
+            .perform(post(API).content(request).contentType(MediaType.APPLICATION_JSON))
+            .andExpect(status().isOk())
+            .andReturn();
+
+    // Validate that the response can be parsed as a valid RunSetStateResponse:
+    RunSetStateResponse response =
+        objectMapper.readValue(
+            result.getResponse().getContentAsString(), RunSetStateResponse.class);
+
+    // Verify database storage:
```

Comment:
the below is mostly copy/pasted from `chainCallsTogether` - there's a lot about the code below that I don't understand.

this is intended to be essentially the same test as `chainCallsTogether`, except with the maximum number of record IDs.

Until we implement the rest of this, it should have the same result. I suppose this test would be updated in the next ticket.

Reply:
no need to apologize -- honestly, a (less well-articulated) version of this thought occurred to me, but I found myself not wanting to engage in a back-and-forth about it, which was a mistake on my part. I should have raised this instead of blindly marching forward.

I think your suggestion for a unit test on the inner function is the best solution. Saloni's upcoming PR should include changes to the original `chainCallsTogether` test such that it takes more than one record ID. It doesn't seem worthwhile to have multiple copies of `chainCallsTogether` for different numbers of inputs.